[MRG+1] Apply the FILES_STORE_S3_ACL setting in ImagesPipeline

jesuslosada opened the pull request:
All images uploaded to S3 using the ImagesPipeline class have the default ACL policy (private), because ImagesPipeline is not respecting the FILES_STORE_S3_ACL setting.

This setting is correctly used in FilesPipeline: https://github.com/scrapy/scrapy/blob/master/scrapy/pipelines/files.py#L258

redapple commented on scrapy/pipelines/images.py:
```diff
@@ -88,6 +88,8 @@ def from_settings(cls, settings):
         s3store = cls.STORE_SCHEMES['s3']
         s3store.AWS_ACCESS_KEY_ID = settings['AWS_ACCESS_KEY_ID']
         s3store.AWS_SECRET_ACCESS_KEY = settings['AWS_SECRET_ACCESS_KEY']
+        s3store.POLICY = settings['FILES_STORE_S3_ACL']
```
I believe the name should be `IMAGES_STORE_S3_ACL` to be consistent with `IMAGES_STORE`. If so, it should also be documented.

jesuslosada replied:
Good point! I thought the name was a reference to the S3FilesStore class, but the setting is only applied in FilesPipeline so it makes sense to create a different setting for ImagesPipeline.

I've just realised that FILES_STORE_S3_ACL is documented on the main settings page, but the other settings related to these pipelines (FILES_STORE, IMAGES_STORE, FILES_URLS_FIELD, etc.) are documented in the media pipeline section. Do you want to move FILES_STORE_S3_ACL (and the new IMAGES_STORE_S3_ACL setting) to the media pipeline section? Or do you prefer to add the new setting in the main settings page too? @redapple @kmike

redapple replied:
I'd go for moving `FILES_STORE_S3_ACL` to `docs/topics/media-pipeline.rst`, and document `IMAGES_STORE_S3_ACL` there too.

jesuslosada commented:
I've added the IMAGES_STORE_S3_ACL setting as previously agreed. I've also updated the documentation. Feel free to suggest any improvements to the documentation.

kmike commented on docs/topics/media-pipeline.rst:
```diff
@@ -146,6 +146,23 @@ Where:
+You can modify the Access Control List (ACL) policy used for the stored files,
+which is defined by the :setting:`FILES_STORE_S3_ACL` and
+:setting:`IMAGES_STORE_S3_ACL` settings. By default, the ACL is set to
+``private``.
```
It'd be nice to explain how to make it public.

jesuslosada commented:
I've added documentation on how to make the files publicly available and a link to the ACLs reference on the Amazon website.

redapple commented:
Thanks a bunch @jesuslosada !

redapple commented:
Needs backporting to 1.1 branch

pawelmhm commented:
Hmm so if in my project I had public-read as access level and I was ok with that your PR will change this to private, right? So all users who were ok with public access will need to set new setting in their projects settings. Wouldn't it be better to keep old setting as default?
